### ISSUE TITLE
CellRenderer sample flash crashes Ruffle

### ISSUE THREAD

**Issue by tsunamistate** [ruffle-rs/ruffle#1199]
Ruffle panics and crashes after about 10 seconds of running when launching the Multiline CellRenderer class example from https://www.adobe.com/support/documentation/en/flash/fl8/samples.html on desktop Ruffle

```
thread 'main' panicked at 'attempt to add with overflow', swf\src\types.rs:109:9
stack backtrace:
   0: backtrace::backtrace::trace_unsynchronized
             at C:\Users\runneradmin\.cargo\registry\src\github.com-1ecc6299db9ec823\backtrace-0.3.46\src\backtrace\mod.rs:66
   1: std::sys_common::backtrace::_print_fmt
             at /rustc/04488afe34512aa4c33566eb16d8c912a3ae04f9\/src\libstd\sys_common\backtrace.rs:78
   2: std::sys_common::backtrace::_print::{{impl}}::fmt
             at /rustc/04488afe34512aa4c33566eb16d8c912a3ae04f9\/src\libstd\sys_common\backtrace.rs:59
   3: core::fmt::write
             at /rustc/04488afe34512aa4c33566eb16d8c912a3ae04f9\/src\libcore\fmt\mod.rs:1076
   4: std::io::Write::write_fmt<std::sys::windows::stdio::Stderr>
             at /rustc/04488afe34512aa4c33566eb16d8c912a3ae04f9\/src\libstd\io\mod.rs:1537
   5: std::sys_common::backtrace::_print
             at /rustc/04488afe34512aa4c33566eb16d8c912a3ae04f9\/src\libstd\sys_common\backtrace.rs:62
   6: std::sys_common::backtrace::print
             at /rustc/04488afe34512aa4c33566eb16d8c912a3ae04f9\/src\libstd\sys_common\backtrace.rs:49
   7: std::panicking::default_hook::{{closure}}
             at /rustc/04488afe34512aa4c33566eb16d8c912a3ae04f9\/src\libstd\panicking.rs:198
   8: std::panicking::default_hook
             at /rustc/04488afe34512aa4c33566eb16d8c912a3ae04f9\/src\libstd\panicking.rs:217
   9: std::panicking::rust_panic_with_hook
             at /rustc/04488afe34512aa4c33566eb16d8c912a3ae04f9\/src\libstd\panicking.rs:526
  10: std::panicking::begin_panic_handler
             at /rustc/04488afe34512aa4c33566eb16d8c912a3ae04f9\/src\libstd\panicking.rs:437
  11: core::panicking::panic_fmt
             at /rustc/04488afe34512aa4c33566eb16d8c912a3ae04f9\/src\libcore\panicking.rs:85
  12: core::panicking::panic
             at /rustc/04488afe34512aa4c33566eb16d8c912a3ae04f9\/src\libcore\panicking.rs:50
  13: swf::types::{{impl}}::add_assign
             at .\swf\src\types.rs:109
  14: ruffle_core::html::dimensions::{{impl}}::add_assign<swf::types::Twips>
             at .\core\src\html\dimensions.rs:79
  15: ruffle_core::html::layout::LayoutContext::newline
             at .\core\src\html\layout.rs:349
  16: ruffle_core::html::layout::LayoutBox::lower_from_text_spans
             at .\core\src\html\layout.rs:706
  17: ruffle_core::display_object::edit_text::EditText::relayout
             at .\core\src\display_object\edit_text.rs:593
  18: ruffle_core::display_object::edit_text::EditText::set_text
             at .\core\src\display_object\edit_text.rs:280
  19: ruffle_core::avm1::globals::text_field::set_text
             at .\core\src\avm1\globals\text_field.rs:42
  20: ruffle_core::avm1::function::Executable::exec
             at .\core\src\avm1\function.rs:241
  21: ruffle_core::avm1::object::script_object::ScriptObject::internal_set
             at .\core\src\avm1\object\script_object.rs:366
  22: ruffle_core::avm1::object::stage_object::{{impl}}::set
             at .\core\src\avm1\object\stage_object.rs:194
  23: ruffle_core::avm1::object::{{impl}}::set
             at .\core\src\avm1\object.rs:45
  24: ruffle_core::avm1::activation::Activation::action_set_member
             at .\core\src\avm1\activation.rs:1770
  25: ruffle_core::avm1::activation::Activation::do_action
             at .\core\src\avm1\activation.rs:529
  26: ruffle_core::avm1::activation::Activation::run_actions
             at .\core\src\avm1\activation.rs:416
  27: ruffle_core::avm1::function::Executable::exec
             at .\core\src\avm1\function.rs:383
  28: ruffle_core::avm1::function::{{impl}}::call
             at .\core\src\avm1\function.rs:555
  29: ruffle_core::avm1::object::{{impl}}::call
             at .\core\src\avm1\object.rs:45
  30: ruffle_core::avm1::value::Value::call
             at .\core\src\avm1\value.rs:491
  31: ruffle_core::avm1::object::TObject::call_method<ruffle_core::avm1::object::stage_object::StageObject>
             at .\core\src\avm1\object.rs:158
  32: ruffle_core::avm1::object::{{impl}}::call_method
             at .\core\src\avm1\object.rs:45
  33: ruffle_core::avm1::activation::Activation::action_call_method
             at .\core\src\avm1\activation.rs:818
  34: ruffle_core::avm1::activation::Activation::do_action
             at .\core\src\avm1\activation.rs:455
  35: ruffle_core::avm1::activation::Activation::run_actions
             at .\core\src\avm1\activation.rs:416
  36: ruffle_core::avm1::function::Executable::exec
             at .\core\src\avm1\function.rs:383
  37: ruffle_core::avm1::function::{{impl}}::call
             at .\core\src\avm1\function.rs:555
  38: ruffle_core::avm1::object::{{impl}}::call
             at .\core\src\avm1\object.rs:45
  39: ruffle_core::avm1::value::Value::call
             at .\core\src\avm1\value.rs:491
  40: ruffle_core::avm1::object::TObject::call_method<ruffle_core::avm1::object::stage_object::StageObject>
             at .\core\src\avm1\object.rs:158
  41: ruffle_core::avm1::object::{{impl}}::call_method
             at .\core\src\avm1\object.rs:45
  42: ruffle_core::avm1::activation::Activation::action_call_method
             at .\core\src\avm1\activation.rs:818
  43: ruffle_core::avm1::activation::Activation::do_action
             at .\core\src\avm1\activation.rs:455
  44: ruffle_core::avm1::activation::Activation::run_actions
             at .\core\src\avm1\activation.rs:416
  45: ruffle_core::avm1::function::Executable::exec
             at .\core\src\avm1\function.rs:383
  46: ruffle_core::avm1::function::{{impl}}::call
             at .\core\src\avm1\function.rs:555
  47: ruffle_core::avm1::object::{{impl}}::call
             at .\core\src\avm1\object.rs:45
  48: ruffle_core::avm1::value::Value::call
             at .\core\src\avm1\value.rs:491
  49: ruffle_core::avm1::object::TObject::call_method<ruffle_core::avm1::object::stage_object::StageObject>
             at .\core\src\avm1\object.rs:158
  50: ruffle_core::avm1::object::{{impl}}::call_method
             at .\core\src\avm1\object.rs:45
  51: ruffle_core::avm1::activation::Activation::action_call_method
             at .\core\src\avm1\activation.rs:818
  52: ruffle_core::avm1::activation::Activation::do_action
             at .\core\src\avm1\activation.rs:455
  53: ruffle_core::avm1::activation::Activation::run_actions
             at .\core\src\avm1\activation.rs:416
  54: ruffle_core::avm1::function::Executable::exec
             at .\core\src\avm1\function.rs:383
  55: ruffle_core::avm1::function::{{impl}}::call
             at .\core\src\avm1\function.rs:555
  56: ruffle_core::avm1::object::{{impl}}::call
             at .\core\src\avm1\object.rs:45
  57: ruffle_core::avm1::value::Value::call
             at .\core\src\avm1\value.rs:491
  58: ruffle_core::avm1::object::super_object::{{impl}}::call_method
             at .\core\src\avm1\object\super_object.rs:137
  59: ruffle_core::avm1::object::{{impl}}::call_method
             at .\core\src\avm1\object.rs:45
  60: ruffle_core::avm1::activation::Activation::action_call_method
             at .\core\src\avm1\activation.rs:818
  61: ruffle_core::avm1::activation::Activation::do_action
             at .\core\src\avm1\activation.rs:455
  62: ruffle_core::avm1::activation::Activation::run_actions
             at .\core\src\avm1\activation.rs:416
  63: ruffle_core::avm1::function::Executable::exec
             at .\core\src\avm1\function.rs:383
  64: ruffle_core::avm1::function::{{impl}}::call
             at .\core\src\avm1\function.rs:555
  65: ruffle_core::avm1::object::{{impl}}::call
             at .\core\src\avm1\object.rs:45
  66: ruffle_core::avm1::value::Value::call
             at .\core\src\avm1\value.rs:491
  67: ruffle_core::avm1::object::super_object::{{impl}}::call_method
             at .\core\src\avm1\object\super_object.rs:137
  68: ruffle_core::avm1::object::{{impl}}::call_method
             at .\core\src\avm1\object.rs:45
  69: ruffle_core::avm1::activation::Activation::action_call_method
             at .\core\src\avm1\activation.rs:818
  70: ruffle_core::avm1::activation::Activation::do_action
             at .\core\src\avm1\activation.rs:455
  71: ruffle_core::avm1::activation::Activation::run_actions
             at .\core\src\avm1\activation.rs:416
  72: ruffle_core::avm1::function::Executable::exec
             at .\core\src\avm1\function.rs:383
  73: ruffle_core::avm1::function::{{impl}}::call
             at .\core\src\avm1\function.rs:555
  74: ruffle_core::avm1::object::{{impl}}::call
             at .\core\src\avm1\object.rs:45
  75: ruffle_core::avm1::value::Value::call
             at .\core\src\avm1\value.rs:491
  76: ruffle_core::avm1::object::TObject::call_method<ruffle_core::avm1::object::stage_object::StageObject>
             at .\core\src\avm1\object.rs:158
  77: ruffle_core::avm1::object::{{impl}}::call_method
             at .\core\src\avm1\object.rs:45
  78: ruffle_core::avm1::activation::Activation::action_call_method
             at .\core\src\avm1\activation.rs:818
  79: ruffle_core::avm1::activation::Activation::do_action
             at .\core\src\avm1\activation.rs:455
  80: ruffle_core::avm1::activation::Activation::run_actions
             at .\core\src\avm1\activation.rs:416
  81: ruffle_core::avm1::function::Executable::exec
             at .\core\src\avm1\function.rs:383
  82: ruffle_core::avm1::function::{{impl}}::call
             at .\core\src\avm1\function.rs:555
  83: ruffle_core::avm1::object::{{impl}}::call
             at .\core\src\avm1\object.rs:45
  84: ruffle_core::avm1::value::Value::call
             at .\core\src\avm1\value.rs:491
  85: ruffle_core::avm1::object::TObject::call_method<ruffle_core::avm1::object::stage_object::StageObject>
             at .\core\src\avm1\object.rs:158
  86: ruffle_core::avm1::object::{{impl}}::call_method
             at .\core\src\avm1\object.rs:45
  87: ruffle_core::avm1::activation::Activation::action_call_method
             at .\core\src\avm1\activation.rs:818
  88: ruffle_core::avm1::activation::Activation::do_action
             at .\core\src\avm1\activation.rs:455
  89: ruffle_core::avm1::activation::Activation::run_actions
             at .\core\src\avm1\activation.rs:416
  90: ruffle_core::avm1::function::Executable::exec
             at .\core\src\avm1\function.rs:383
  91: ruffle_core::avm1::function::{{impl}}::call
             at .\core\src\avm1\function.rs:555
  92: ruffle_core::avm1::object::{{impl}}::call
             at .\core\src\avm1\object.rs:45
  93: ruffle_core::avm1::value::Value::call
             at .\core\src\avm1\value.rs:491
  94: ruffle_core::avm1::object::TObject::call_method<ruffle_core::avm1::object::stage_object::StageObject>
             at .\core\src\avm1\object.rs:158
  95: ruffle_core::avm1::object::{{impl}}::call_method
             at .\core\src\avm1\object.rs:45
  96: ruffle_core::avm1::activation::Activation::action_call_method
             at .\core\src\avm1\activation.rs:818
  97: ruffle_core::avm1::activation::Activation::do_action
             at .\core\src\avm1\activation.rs:455
  98: ruffle_core::avm1::activation::Activation::run_actions
             at .\core\src\avm1\activation.rs:416
  99: ruffle_core::avm1::function::Executable::exec
             at .\core\src\avm1\function.rs:383
 100: ruffle_core::avm1::function::{{impl}}::call
             at .\core\src\avm1\function.rs:555
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
error: process didn't exit successfully: `target\debug\ruffle_desktop.exe D:\swf\CellRenderer_tutorial.swf` (exit code: 0xc000001d, STATUS_ILLEGAL_INSTRUCTION)
```

[MultiLineCellRenderer.zip](https://github.com/ruffle-rs/ruffle/files/5247860/MultiLineCellRenderer.zip)

- Windows 10 Pro x64
- rustc 1.46.0
- commit 6f2655cf95d5c2185492d8d743bc8bc721e8ca34

**Comment by adrian17** [comment]
Might be related to https://github.com/ruffle-rs/ruffle/issues/3088 .

**Comment by adrian17** [comment]
Should be fixed by https://github.com/ruffle-rs/ruffle/pull/4339 .

**Comment by tsunamistate** [comment]
Fix confirmed on 30f088d16f871a9f279f4c540bd60fa62cc634d5
